pyomo/gdp/plugins/chull.py

```diff
@@ -840,5 +853,131 @@ def _xform_constraint(self, obj, disjunct, infodict, var_substitute_map,
+                    else:
+                        constraintMap['transformedConstraints'][
+                            c] = [newConstraint[i,'ub']]
+                    constraintMap['srcConstraints'][newConstraint[i,'ub']] = c
                 else:
                     newConstraint.add('ub', newConsExpr)
+                    transformed = constraintMap['transformedConstraints'].get(c)
+                    if transformed is not None:
+                        transformed.append(newConstraint['ub'])
+                    else:
+                        constraintMap['transformedConstraints'][
+                            c] = [newConstraint['ub']]
+                    constraintMap['srcConstraints'][newConstraint['ub']] = c
+
+        # deactivate now that we have transformed
+        obj.deactivate()
+
+    def get_src_disjunct(self, transBlock):
+        return get_src_disjunct(transBlock)
+
+    def get_src_disjunction(self, xor_constraint):
+        return get_src_disjunction(xor_constraint)
+
+    def get_src_constraint(self, transformedConstraint):
+        return get_src_constraint(transformedConstraint)
+
+    def get_transformed_constraints(self, srcConstraint):
+        return get_transformed_constraints(srcConstraint)
+
+    def get_disaggregated_var(self, v, disjunct):
+        """
+        Returns the disaggregated variable corresponding to the Var v and the
+        Disjunct disjunct.
+
+        If v is a local variable, this method will return v.
+
+        Parameters
+        ----------
+        v: a Var which appears in a constraint in a transformed Disjunct
+        disjunct: a transformed Disjunct in which v appears
+        """
+        if disjunct._transformation_block is None:
+            raise GDP_Error("Disjunct %s has not been transformed" 
+                            % disjunct.name)
+        transBlock = disjunct._transformation_block()
+        try:
+            return transBlock._disaggregatedVarMap['disaggregatedVar'][v]
+        except:
+            raise GDP_Error("It does not appear %s is a "
+                            "variable which appears in disjunct %s"
+                            % (v.name, disjunct.name))
+        
+    def get_src_var(self, disaggregated_var):
+        """
+        Returns the original model variable to which disaggregated_var 
+        corresponds.
+
+        Parameters
+        ----------
+        disaggregated_var: a Var which was created by the chull 
+                           transformation as a disaggregated variable 
+                           (and so appears on a transformation block 
+                           of some Disjunct)
+        """
+        transBlock = disaggregated_var.parent_block()
+        try:
+            src_disjunct = transBlock._srcDisjunct()
+        except:
+            raise GDP_Error("%s does not appear to be a disaggregated variable"
+                            % disaggregated_var.name)
```

Comment:
This is dangerous as it can hide underlying bugs.  Either catch a specific exception, or (probably better) log this message as an error and re-raise the original exeception.